app/component/IndexPage.js

```diff
@@ -280,6 +302,17 @@ class IndexPage extends React.Component {
       'Stops',
     ];
 
+    if (
+      config.startSearchFromUserLocation &&
+      currentLocation &&
+      config.areaPolygon
+    ) {
+      const originPoint = [currentLocation.lon, currentLocation.lat];
+      if (inside(originPoint, config.areaPolygon)) {
+        this.context.executeAction(storeOrigin, currentLocation);
```

Comment:
render should not have any side effects. Debugger gives warnings about cascading updates. Maybe this  code should be in componentDidUpdate.

Reply:
Absolutely!